# <h1 style="text-align: center; "><strong>Proceso de Extracción, Transformación y Carga</strong></h1>

Para el desarrollo de las fases I  y II del proyecto seguiremos una metodología basada en el Análisis Exploratorio de Datos (EDA). Para este procedimiento realizaremos las siguientes actividades:

1. Carga/Extracción de los datos.
2. Exploración inicial.
3. Limpieza y preparación.
4. Manipulación y Carga de Datos.

### 1. Carga/Extracción de los datos

Importamos las librerías que se usarán en el proceso de ETL

In [167]:
from datetime import date
import pandas as pd

Insertamos los datos del Dataset "desaparecidos_colombia_1930_a_2025.csv" en un Dataframe para facilitar su manipulación. Este Dataset es público y fue obtenido en la página <https://www.datos.gov.co/Justicia-y-Derecho/Desaparecidos-Colombia-hist-rico-a-os-1930-a-mayo-/8hqm-7fdt/about_data> y fue suministrado por el Instituto Nacional de Medicina Legal y Ciencias Forenses.

In [168]:
data_pre = pd.read_csv("Datasets/desaparecidos_colombia_1930_a_2025.csv")

### 2. Exploración inicial

Utilizamos algunos métodos sencillos para tener una idea general de contenido del Dataset


In [169]:
data_pre.head(10)

ID Entidad que realiza el registro de la desaparición  \
0   1  Instituto Nacional de Medicina Legal y Ciencia...   
1   2  Instituto Nacional de Medicina Legal y Ciencia...   
2   3  Instituto Nacional de Medicina Legal y Ciencia...   
3   4  Instituto Nacional de Medicina Legal y Ciencia...   
4   5  Instituto Nacional de Medicina Legal y Ciencia...   
5   6  Instituto Nacional de Medicina Legal y Ciencia...   
6   7  Instituto Nacional de Medicina Legal y Ciencia...   
7   8  Instituto Nacional de Medicina Legal y Ciencia...   
8   9  Instituto Nacional de Medicina Legal y Ciencia...   
9  10  Instituto Nacional de Medicina Legal y Ciencia...   

  Estado de la desaparición Clasificación de la desaparición  \
0              Desaparecido                  Sin información   
1              Desaparecido                  Sin información   
2              Desaparecido                  Sin información   
3             Aparecio Vivo                  Sin información   
4             Aparecio Vivo                  Sin información   
5              Desaparecido                  Sin información   
6              Desaparecido                  Sin información   
7              Desaparecido                  Sin información   
8              Desaparecido                  Sin información   
9              Desaparecido                  Sin información   

  Sexo del desaparecido Grupo mayor y menor de edad del desaparecido  \
0                Hombre                b) Mayores de Edad (>18 años)   
1                Hombre                b) Mayores de Edad (>18 años)   
2                 Mujer                a) Menores de Edad (<18 años)   
3                 Mujer                a) Menores de Edad (<18 años)   
4                Hombre                b) Mayores de Edad (>18 años)   
5                 Mujer                b) Mayores de Edad (>18 años)   
6                Hombre                b) Mayores de Edad (>18 años)   
7                Hombre                b) Mayores de Edad (>18 años)   
8                 Mujer                a) Menores de Edad (<18 años)   
9                 Mujer                a) Menores de Edad (<18 años)   

  Grupo de edad quinquenal del desaparecido  \
0                                 (20 a 24)   
1                                 (20 a 24)   
2                                 (10 a 14)   
3                                 (15 a 17)   
4                                 (70 a 74)   
5                                 (20 a 24)   
6                                 (20 a 24)   
7                                 (35 a 39)   
8                                 (15 a 17)   
9                                 (10 a 14)   

  Grupo de edad judicial del desaparecido Ciclo vital del desaparecido  \
0                               (20 a 24)           (18 a 28) Juventud   
1                               (20 a 24)           (18 a 28) Juventud   
2                               (10 a 13)       (12 a 17) Adolescencia   
3                               (14 a 17)       (12 a 17) Adolescencia   
4                               (70 a 74)      (60 y más) Adulto mayor   
5                               (20 a 24)           (18 a 28) Juventud   
6                               (20 a 24)           (18 a 28) Juventud   
7                               (35 a 39)            (29 a 59) Adultez   
8                               (14 a 17)       (12 a 17) Adolescencia   
9                               (10 a 13)       (12 a 17) Adolescencia   

  Estado civil del desaparecido  ... Mes de la desaparición  \
0                    Soltero(a)  ...              diciembre   
1                    Soltero(a)  ...              diciembre   
2                    Soltero(a)  ...              diciembre   
3                    Soltero(a)  ...              diciembre   
4                   Unión libre  ...              diciembre   
5                    Soltero(a)  ...                  enero   
6                    Soltero(a)  ...                octubre   
7      

In [170]:
data_pre.shape

(195808, 30)

In [171]:
data_pre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195808 entries, 0 to 195807
Data columns (total 30 columns):
 #   Column                                              Non-Null Count   Dtype  
---  ------                                              --------------   -----  
 0   ID                                                  195808 non-null  int64  
 1   Entidad que realiza el registro de la desaparición  195808 non-null  object 
 2   Estado de la desaparición                           195808 non-null  object 
 3   Clasificación de la desaparición                    195808 non-null  object 
 4   Sexo del desaparecido                               195808 non-null  object 
 5   Grupo mayor y menor de edad del desaparecido        195808 non-null  object 
 6   Grupo de edad quinquenal del desaparecido           195808 non-null  object 
 7   Grupo de edad judicial del desaparecido             195808 non-null  object 
 8   Ciclo vital del desaparecido                        195808 non-n

In [172]:
data_pre.describe()

ID  Codigo Dane Departamento  Codigo Dane Municipio
count  195808.000000             194487.000000          194487.000000
mean    97904.500000                 36.384699           32824.216513
std     56525.045092                 65.354088           28040.348294
min         1.000000                  5.000000             999.000000
25%     48952.750000                 11.000000           11001.000000
50%     97904.500000                 17.000000           17388.000000
75%    146856.250000                 54.000000           54405.000000
max    195808.000000                999.000000           99773.000000

Se observó que el Dataset contiene información personal y de la desaparición de individuos de diferentes nacionalidades en colombia y otros países durante el periodo comprendido entre 1930 a 2025. El Dataset original tiene 195808 registros de desaparecidos y 30 columnas que representan la información disponible sobre el individuo. Inicialmente se tienen la siguiente información sobre las características:

- 1 característica de tipo entero, 2 características de tipo flotante, y 27 características de tipo object
- Las características "Codigo Dane Departamento" y "Codigo Dane Municipio" son las únicas características con datos faltantes. Sin embargo, es una cantidad mínima comparado con la cantidad de registros totales
- Algunas columnas podrían ser redundantes. Por ejemplo, se tienen columnas de Fecha, Año, Mes y Dia de la desaparición. Otro ejemplo es que tenemos el código Dane del Municipio y Departamento, y además tenemos el nombre del Municipio y Departamento.


### 3. Limpieza y preparación


Inicialmente reordenamos las columnas del Dataframe con el fin de tener una mejor organización de la información

In [173]:
data_pre = data_pre[[
    "ID",
    "Fecha de la desaparición",
    "Año de la desaparición",
    "Mes de la desaparición",
    "Día de la desaparición",
    "Estado de la desaparición",
    "Clasificación de la desaparición",
    "Entidad que realiza el registro de la desaparición",
    "País donde ocurre la desaparición",
    "Departamento donde ocurre la desaparición DANE",
    "Codigo Dane Departamento",
    "Municipio donde ocurre la desaparición DANE",
    "Codigo Dane Municipio", 
    "Zona donde ocurre la desaparición",
    "Localidad donde ocurre la desaparición",
    "Sexo del desaparecido",
    "País de nacimiento del desaparecido",
    "Estado civil del desaparecido",
    "Grupo mayor y menor de edad del desaparecido",
    "Grupo de edad quinquenal del desaparecido",
    "Grupo de edad judicial del desaparecido",
    "Ciclo vital del desaparecido",
    "Escolaridad del desaparecido",
    "Pertenencia étnica del desaparecido",
    "Pueblo indígena del desaparecido",
    "Pertenencia grupal del desaparecido",
    "Identidad de género del desaparecido",
    "Orientación sexual del desaparecido",
    "Transgénero",
    "Contexto"
]]

In [174]:
print(data_pre["ID"].head(10))
print(data_pre["ID"].value_counts())

data_pre.drop(columns = "ID", inplace = True)
# Se elimina la columna

0     1
1     2
2     3
3     4
4     5
5     6
6     7
7     8
8     9
9    10
Name: ID, dtype: int64
ID
1         1
2         1
3         1
4         1
5         1
         ..
195804    1
195805    1
195806    1
195807    1
195808    1
Name: count, Length: 195808, dtype: int64


In [175]:
print(data_pre["Fecha de la desaparición"].head(10))

def parseDate(x):
    return date(int(x[0:4]), int(x[5:7]), int(x[8:10]))
# Creamos una función para extraer el año, mes y dia

data_pre["Fecha de la desaparición"] = pd.to_datetime(data_pre["Fecha de la desaparición"].apply(lambda x: parseDate(x)))
# Convertimos los datos de la columna a tipo Date (Fecha)

0    2006-12-29T00:00:00.000
1    2006-12-26T00:00:00.000
2    2006-12-29T00:00:00.000
3    2006-12-29T00:00:00.000
4    2006-12-31T00:00:00.000
5    2007-01-01T00:00:00.000
6    2006-10-17T00:00:00.000
7    1900-01-01T00:00:00.000
8    2006-12-31T00:00:00.000
9    2006-12-30T00:00:00.000
Name: Fecha de la desaparición, dtype: object


In [176]:
print(data_pre["Año de la desaparición"].head(10))
print(data_pre["Año de la desaparición"].value_counts())

data_pre["Año de la desaparición"] = data_pre["Fecha de la desaparición"].apply(lambda x: int(x.year))
# La columna "Año de la desaparición" se llena con el año de la fecha para llenar los datos sin información

data_pre = data_pre[data_pre["Año de la desaparición"] >= 1986]
data_pre = data_pre[data_pre["Año de la desaparición"] <= 2024]
# Se obtienen las desapariciones entre el año 1986 y 2024

0               2006
1               2006
2               2006
3               2006
4               2006
5               2007
6               2006
7    Sin información
8               2006
9               2006
Name: Año de la desaparición, dtype: object
Año de la desaparición
2011    8431
2012    8365
2013    8074
2010    7917
2015    7800
        ... 
1953       2
1930       1
1937       1
1948       1
1938       1
Name: count, Length: 84, dtype: int64


In [177]:
print(data_pre["Mes de la desaparición"].head(10))

data_pre.drop(columns = "Mes de la desaparición", inplace = True)
# Se elimina la columna

0     diciembre
1     diciembre
2     diciembre
3     diciembre
4     diciembre
5         enero
6       octubre
8     diciembre
9     diciembre
10    diciembre
Name: Mes de la desaparición, dtype: object


In [178]:
print(data_pre["Día de la desaparición"].value_counts())

def getFullWeekday(x):
    if x.weekday() == 0:
        return "Lunes"
    elif x.weekday() == 1:
        return "Martes"
    elif x.weekday() == 2:
        return "Miércoles"
    elif x.weekday() == 3:
        return "Jueves"
    elif x.weekday() == 4:
        return "Viernes"
    elif x.weekday() == 5:
        return "Sábado"
    elif x.weekday() == 6:
        return "Domingo"
# Se crea una función para obtener el dia de la semana a partir de la fecha

data_pre["Día de la desaparición"] = data_pre["Fecha de la desaparición"].apply(lambda x: getFullWeekday(x))
# Se aplica la función a la columna

Día de la desaparición
sábado       31122
viernes      28019
lunes        28001
martes       26491
domingo      25628
miércoles    25454
jueves       25219
Name: count, dtype: int64


In [179]:
print(data_pre["Estado de la desaparición"].head(10))
print(data_pre["Estado de la desaparición"].value_counts())

data_pre["Estado de la desaparición"].replace({
    "Aparecio Vivo": "Aparecido Vivo",
    "Aparecio Muerto": "Aparecido Muerto"
})
# Se cambian algunos datos

0      Desaparecido
1      Desaparecido
2      Desaparecido
3     Aparecio Vivo
4     Aparecio Vivo
5      Desaparecido
6      Desaparecido
8      Desaparecido
9      Desaparecido
10     Desaparecido
Name: Estado de la desaparición, dtype: object
Estado de la desaparición
Desaparecido       121836
Aparecio Vivo       57453
Aparecio Muerto     10645
Name: count, dtype: int64


0           Desaparecido
1           Desaparecido
2           Desaparecido
3         Aparecido Vivo
4         Aparecido Vivo
               ...      
195802      Desaparecido
195803      Desaparecido
195805      Desaparecido
195806      Desaparecido
195807      Desaparecido
Name: Estado de la desaparición, Length: 189934, dtype: object

In [180]:
print(data_pre["Clasificación de la desaparición"].head(10))
print(data_pre["Clasificación de la desaparición"].value_counts())

0     Sin información
1     Sin información
2     Sin información
3     Sin información
4     Sin información
5     Sin información
6     Sin información
8     Sin información
9     Sin información
10    Sin información
Name: Clasificación de la desaparición, dtype: object
Clasificación de la desaparición
Sin información                       134867
Desaparicion Presuntamente Forzada     55067
Name: count, dtype: int64


In [181]:
print(data_pre["Entidad que realiza el registro de la desaparición"].head(10))
print(data_pre["Entidad que realiza el registro de la desaparición"].value_counts())

data_pre.drop(columns = "Entidad que realiza el registro de la desaparición", inplace = True)
# Se elimina la columna

0     Instituto Nacional de Medicina Legal y Ciencia...
1     Instituto Nacional de Medicina Legal y Ciencia...
2     Instituto Nacional de Medicina Legal y Ciencia...
3     Instituto Nacional de Medicina Legal y Ciencia...
4     Instituto Nacional de Medicina Legal y Ciencia...
5     Instituto Nacional de Medicina Legal y Ciencia...
6     Instituto Nacional de Medicina Legal y Ciencia...
8     Instituto Nacional de Medicina Legal y Ciencia...
9     Instituto Nacional de Medicina Legal y Ciencia...
10    Instituto Nacional de Medicina Legal y Ciencia...
Name: Entidad que realiza el registro de la desaparición, dtype: object
Entidad que realiza el registro de la desaparición
Fiscalia - Cuerpo Tecnico Investigacion                     73035
Instituto Nacional de Medicina Legal y Ciencias Forenses    69795
Fiscalia General de la Nacion                               15906
Fiscalia - Unidad Justicia y Paz                            11802
Policia Nacional                                     

In [182]:
print(data_pre["País donde ocurre la desaparición"].head(10))
print(data_pre["País donde ocurre la desaparición"].value_counts())

print("Tamaño antes:", data_pre["País de nacimiento del desaparecido"].shape)

data_pre = data_pre[data_pre["País donde ocurre la desaparición"] == "Colombia"]
# Se filtra para las personas que desaparecieron en Colombia

print("Tamaño después:", data_pre["País de nacimiento del desaparecido"].shape)

data_pre.drop(columns = "País donde ocurre la desaparición", inplace = True)
# Se elimina la columna

0     Colombia
1     Colombia
2     Colombia
3     Colombia
4     Colombia
5     Colombia
6     Colombia
8     Colombia
9     Colombia
10    Colombia
Name: País donde ocurre la desaparición, dtype: object
País donde ocurre la desaparición
Colombia                188639
Venezuela                  447
Ecuador                    257
México                     156
Perú                        71
Sin información             50
Guatemala                   49
Panamá                      44
Ucrania                     41
Estados Unidos              31
Chile                       25
Brasil                      23
España                      16
Bolivia                     12
Honduras                     9
Francia                      6
Alemania                     6
Costa Rica                   6
Italia                       6
República Dominicana         5
Aruba                        4
China                        3
Argentina                    3
Bahamas                      3
Paraguay         

In [183]:
print(data_pre["Departamento donde ocurre la desaparición DANE"].head(10))
print(data_pre["Departamento donde ocurre la desaparición DANE"].value_counts())

0     Bogotá, D.C.
1     Bogotá, D.C.
2     Bogotá, D.C.
3     Bogotá, D.C.
4     Bogotá, D.C.
5     Bogotá, D.C.
6     Bogotá, D.C.
8     Bogotá, D.C.
9     Bogotá, D.C.
10    Bogotá, D.C.
Name: Departamento donde ocurre la desaparición DANE, dtype: object
Departamento donde ocurre la desaparición DANE
Bogotá, D.C.                                                56343
Antioquia                                                   24887
Valle del Cauca                                             17122
Meta                                                         7868
Cundinamarca                                                 7442
Norte de Santander                                           6640
Risaralda                                                    5661
Caldas                                                       5383
Nariño                                                       4911
Tolima                                                       4532
Santander                          

In [184]:
print(data_pre["Codigo Dane Departamento"].head(10))
print(data_pre["Codigo Dane Departamento"].value_counts())

data_pre["Codigo Dane Departamento"] = data_pre["Codigo Dane Departamento"].apply(lambda x: int(x))
# Se cambia de tipo float a int el código Dane del Departamento

0     11.0
1     11.0
2     11.0
3     11.0
4     11.0
5     11.0
6     11.0
8     11.0
9     11.0
10    11.0
Name: Codigo Dane Departamento, dtype: float64
Codigo Dane Departamento
11.0     56343
5.0      24887
76.0     17122
50.0      7868
25.0      7442
54.0      6640
66.0      5661
17.0      5383
52.0      4911
73.0      4532
68.0      4265
47.0      4200
8.0       3358
20.0      3328
41.0      3257
86.0      2958
18.0      2938
15.0      2915
13.0      2550
95.0      2518
23.0      2351
19.0      2191
27.0      2087
63.0      2059
85.0      1798
81.0      1559
70.0       969
44.0       919
999.0      626
99.0       547
91.0       128
88.0       121
94.0       117
97.0        91
Name: count, dtype: int64


In [185]:
print(data_pre["Municipio donde ocurre la desaparición DANE"].head(10))
print(data_pre["Municipio donde ocurre la desaparición DANE"].value_counts())

0     Bogotá, D.C.
1     Bogotá, D.C.
2     Bogotá, D.C.
3     Bogotá, D.C.
4     Bogotá, D.C.
5     Bogotá, D.C.
6     Bogotá, D.C.
8     Bogotá, D.C.
9     Bogotá, D.C.
10    Bogotá, D.C.
Name: Municipio donde ocurre la desaparición DANE, dtype: object
Municipio donde ocurre la desaparición DANE
Bogotá, D.C.       56343
Medellín            9944
Cali                9888
Pereira             3717
Cúcuta              3576
                   ...  
El Cocuy               1
Pana Pana              1
Mirití - Paraná        1
Vetas                  1
Panqueba               1
Name: count, Length: 1015, dtype: int64


In [186]:
print(data_pre["Codigo Dane Municipio"].head(10))
print(data_pre["Codigo Dane Municipio"].value_counts())

data_pre.drop(columns = "Codigo Dane Municipio", inplace = True)
# Se elimina la columna

0     11001.0
1     11001.0
2     11001.0
3     11001.0
4     11001.0
5     11001.0
6     11001.0
8     11001.0
9     11001.0
10    11001.0
Name: Codigo Dane Municipio, dtype: float64
Codigo Dane Municipio
11001.0    56343
5001.0      9944
76001.0     9888
66001.0     3717
54001.0     3576
           ...  
52506.0        1
15244.0        1
15215.0        1
15162.0        1
15822.0        1
Name: count, Length: 1097, dtype: int64


In [187]:
print(data_pre["Zona donde ocurre la desaparición"].head(10))
print(data_pre["Zona donde ocurre la desaparición"].value_counts())

0     Cabecera municipal
1     Cabecera municipal
2     Cabecera municipal
3     Cabecera municipal
4     Cabecera municipal
5     Cabecera municipal
6     Cabecera municipal
8     Cabecera municipal
9     Cabecera municipal
10    Cabecera municipal
Name: Zona donde ocurre la desaparición, dtype: object
Zona donde ocurre la desaparición
Cabecera municipal                                                 133524
Parte rural (vereda y campo)                                        30982
Sin información                                                     12439
Centro poblado (corregimiento, inspección de policía y caserío)     11694
Name: count, dtype: int64


In [188]:
print(data_pre["Localidad donde ocurre la desaparición"].head(10))
print(data_pre["Localidad donde ocurre la desaparición"].value_counts())

data_pre.drop(columns = "Localidad donde ocurre la desaparición", inplace = True)
# Se elimina la columna

0     Sin información
1             Kennedy
2            Engativá
3             Kennedy
4     Sin información
5       San Cristóbal
6       San Cristóbal
8            Engativá
9      Ciudad Bolívar
10            Kennedy
Name: Localidad donde ocurre la desaparición, dtype: object
Localidad donde ocurre la desaparición
Sin información       138624
Kennedy                 6042
Ciudad Bolívar          5747
Bosa                    4393
San Cristóbal           4172
Suba                    3947
Engativá                3755
Usme                    3286
Rafael Uribe Uribe      2740
Santafé                 2668
Los Mártires            2131
Usaquén                 2083
Tunjuelito              1936
Fontibón                1820
Puente Aranda           1366
Chapinero               1312
Antonio Nariño           890
Barrios Unidos           674
Teusaquillo              580
La Candelaria            436
Sumapaz                   37
Name: count, dtype: int64


In [189]:
print(data_pre["Sexo del desaparecido"].head(10))
print(data_pre["Sexo del desaparecido"].value_counts())

0     Hombre
1     Hombre
2      Mujer
3      Mujer
4     Hombre
5      Mujer
6     Hombre
8      Mujer
9      Mujer
10    Hombre
Name: Sexo del desaparecido, dtype: object
Sexo del desaparecido
Hombre    131404
Mujer      57235
Name: count, dtype: int64


In [190]:
print(data_pre["País de nacimiento del desaparecido"].head(10))
print(data_pre["País de nacimiento del desaparecido"].value_counts())

print("Tamaño antes:", data_pre["País de nacimiento del desaparecido"].shape)

data_pre = data_pre[data_pre["País de nacimiento del desaparecido"] == "Colombia"]
# Se filtra para personas nacidas en Colombia

print("Tamaño después:" ,data_pre["País de nacimiento del desaparecido"].shape, "\n")

data_pre.drop(columns = "País de nacimiento del desaparecido", inplace = True)
# Se elimina la columna

0     Sin información
1            Colombia
2            Colombia
3     Sin información
4            Colombia
5            Colombia
6            Colombia
8            Colombia
9            Colombia
10           Colombia
Name: País de nacimiento del desaparecido, dtype: object
País de nacimiento del desaparecido
Colombia                  172952
Sin información            11682
Venezuela                   3342
Ecuador                      143
Estados Unidos                86
                           ...  
Trinidad y Tobago              1
Israel                         1
Emiratos Árabes Unidos         1
Japón                          1
Congo (Rep. Dem.)              1
Name: count, Length: 74, dtype: int64
Tamaño antes: (188639,)
Tamaño después: (172952,) 



In [191]:
print(data_pre["Estado civil del desaparecido"].head(10))
print(data_pre["Estado civil del desaparecido"].value_counts())

1          Soltero(a)
2          Soltero(a)
4         Unión libre
5          Soltero(a)
6          Soltero(a)
8          Soltero(a)
9          Soltero(a)
10    Sin información
11         Soltero(a)
12         Soltero(a)
Name: Estado civil del desaparecido, dtype: object
Estado civil del desaparecido
Soltero(a)                    105167
Unión libre                    28954
Sin información                16614
Casado(a)                      12937
Separado(a), divorciado(a)      5071
No aplica                       2585
Viudo(a)                        1624
Name: count, dtype: int64


In [192]:
print(data_pre["Grupo mayor y menor de edad del desaparecido"].head(10))
print(data_pre["Grupo mayor y menor de edad del desaparecido"].value_counts())

data_pre.drop(columns = "Grupo mayor y menor de edad del desaparecido", inplace = True)
# Se elimina la columna

1     b) Mayores de Edad (>18 años)
2     a) Menores de Edad (<18 años)
4     b) Mayores de Edad (>18 años)
5     b) Mayores de Edad (>18 años)
6     b) Mayores de Edad (>18 años)
8     a) Menores de Edad (<18 años)
9     a) Menores de Edad (<18 años)
10    b) Mayores de Edad (>18 años)
11    a) Menores de Edad (<18 años)
12    b) Mayores de Edad (>18 años)
Name: Grupo mayor y menor de edad del desaparecido, dtype: object
Grupo mayor y menor de edad del desaparecido
b) Mayores de Edad (>18 años)    126388
a) Menores de Edad (<18 años)     46564
Name: count, dtype: int64


In [193]:
print(data_pre["Grupo de edad quinquenal del desaparecido"].head(10))
print(data_pre["Grupo de edad quinquenal del desaparecido"].value_counts())

data_pre.drop(columns = "Grupo de edad quinquenal del desaparecido", inplace = True)
# Se elimina la columna

1     (20 a 24)
2     (10 a 14)
4     (70 a 74)
5     (20 a 24)
6     (20 a 24)
8     (15 a 17)
9     (10 a 14)
10    (40 a 44)
11    (15 a 17)
12    (25 a 29)
Name: Grupo de edad quinquenal del desaparecido, dtype: object
Grupo de edad quinquenal del desaparecido
(15 a 17)     24687
(20 a 24)     23851
(25 a 29)     21766
(10 a 14)     19292
(30 a 34)     17434
(35 a 39)     13755
(40 a 44)     10380
(18 a 19)      9318
(45 a 49)      7921
(50 a 54)      6578
(55 a 59)      4482
(60 a 64)      3297
(65 a 69)      2353
(80 y más)     1850
(70 a 74)      1791
(75 a 79)      1612
(05 a 09)      1446
(00 a 04)      1139
Name: count, dtype: int64


In [194]:
print(data_pre["Grupo de edad judicial del desaparecido"].head(10))
print(data_pre["Grupo de edad judicial del desaparecido"].value_counts())

def formatAge(x):
    x1 = x.split("(")
    x2 = x1[1].split(")")
    return f"{x2[0]} Años"
# Se crea una función para obtener el grupo de edad y llevarlo a un formato común

data_pre["Grupo de edad judicial del desaparecido"] = data_pre["Grupo de edad judicial del desaparecido"].apply(lambda x: formatAge(x))
# Se aplica la función a la columna "Grupo de edad judicial del desaparecido"

1     (20 a 24)
2     (10 a 13)
4     (70 a 74)
5     (20 a 24)
6     (20 a 24)
8     (14 a 17)
9     (10 a 13)
10    (40 a 44)
11    (14 a 17)
12    (25 a 28)
Name: Grupo de edad judicial del desaparecido, dtype: object
Grupo de edad judicial del desaparecido
(14 a 17)     33323
(20 a 24)     23851
(29 a 34)     21350
(25 a 28)     17850
(35 a 39)     13755
(10 a 13)     10656
(40 a 44)     10380
(18 a 19)      9318
(45 a 49)      7921
(50 a 54)      6578
(55 a 59)      4482
(60 a 64)      3297
(65 a 69)      2353
(80 y más)     1850
(70 a 74)      1791
(75 a 79)      1612
(05 a 09)      1446
(00 a 04)      1139
Name: count, dtype: int64


In [195]:
print(data_pre["Ciclo vital del desaparecido"].head(10))
print(data_pre["Ciclo vital del desaparecido"].value_counts())

def getLifeCicle(x):
    range_ = x.split(" ")
    try: 
        inf = int(range_[0])
        sup = int(range_[2])
    except:
        inf = int(range_[0])
        sup = 80

    age = (inf + sup) / 2

    if (age < 6):
        return "Primera infancia (de 0 a 5 años)"
    elif (age >= 6) and (age < 12):
        return "Infancia (de 6 a 11 años)"
    elif (age >= 12) and (age < 18):
        return "Adolescencia (de 12 a 17 años)"
    elif (age >= 18) and (age < 29):
        return "Juventud (de 18 a 28 años)"
    elif (age >= 29) and (age < 60):
        return "Adultez (de 29 a 59 años)"
    elif (age >= 60):
        return "Adulto Mayor (mayor de 60 años)"
# Se crea una función para obtener el ciclo vital del individuo

data_pre["Ciclo vital del desaparecido"] = data_pre["Grupo de edad judicial del desaparecido"].apply(lambda x: getLifeCicle(x))
# Se aplica la función a la columna "Ciclo vital del desaparecido"

1          (18 a 28) Juventud
2      (12 a 17) Adolescencia
4     (60 y más) Adulto mayor
5          (18 a 28) Juventud
6          (18 a 28) Juventud
8      (12 a 17) Adolescencia
9      (12 a 17) Adolescencia
10          (29 a 59) Adultez
11     (12 a 17) Adolescencia
12         (18 a 28) Juventud
Name: Ciclo vital del desaparecido, dtype: object
Ciclo vital del desaparecido
(29 a 59) Adultez             64466
(18 a 28) Juventud            51019
(12 a 17) Adolescencia        42227
(60 y más) Adulto mayor       10903
(06 a 11) Infancia             2996
(00 a 05) Primera Infancia     1341
Name: count, dtype: int64


In [196]:
print(data_pre["Escolaridad del desaparecido"].head(10))
print(data_pre["Escolaridad del desaparecido"].value_counts())

1     Educación básica secundaria o secundaria baja
2     Educación básica secundaria o secundaria baja
4                         Educación básica primaria
5                         Educación básica primaria
6     Educación básica secundaria o secundaria baja
8                 Educación media o secundaria alta
9     Educación básica secundaria o secundaria baja
10                        Educación básica primaria
11                Educación media o secundaria alta
12                        Educación básica primaria
Name: Escolaridad del desaparecido, dtype: object
Escolaridad del desaparecido
Educación básica primaria                        70494
Educación básica secundaria o secundaria baja    30572
Sin información                                  27963
Educación media o secundaria alta                18265
Educación inicial y educación preescolar         11476
Sin escolaridad                                   8121
Educación técnica profesional y tecnológica       3743
Universitario   

In [197]:
print(data_pre["Pertenencia étnica del desaparecido"].head(10))
print(data_pre["Pertenencia étnica del desaparecido"].value_counts())

1     Sin pertenencia étnica
2     Sin pertenencia étnica
4     Sin pertenencia étnica
5     Sin pertenencia étnica
6     Sin pertenencia étnica
8     Sin pertenencia étnica
9     Sin pertenencia étnica
10    Sin pertenencia étnica
11    Sin pertenencia étnica
12    Sin pertenencia étnica
Name: Pertenencia étnica del desaparecido, dtype: object
Pertenencia étnica del desaparecido
Sin pertenencia étnica    171099
Indígena                    1159
Negro/Afrodescendiente       508
Grupos étnicos               123
Sin información               45
Raizal                        13
Ninguna                        3
Rrom(Gitanos)                  2
Name: count, dtype: int64


In [198]:
print(data_pre["Pueblo indígena del desaparecido"].head(10))
print(data_pre["Pueblo indígena del desaparecido"].value_counts())

data_pre.drop(columns = "Pueblo indígena del desaparecido", inplace = True)
# Se elimina la columna

1     Ninguno
2     Ninguno
4     Ninguno
5     Ninguno
6     Ninguno
8     Ninguno
9     Ninguno
10    Ninguno
11    Ninguno
12    Ninguno
Name: Pueblo indígena del desaparecido, dtype: object
Pueblo indígena del desaparecido
Ninguno              171793
Indígena                477
Nasa - Páez             116
Emberá                   76
Awa                      48
                      ...  
Eperara Siapidara         1
Paez                      1
Guayabero                 1
Uwa                       1
Curripaco                 1
Name: count, Length: 95, dtype: int64


In [199]:
print(data_pre["Pertenencia grupal del desaparecido"].head(10))
print(data_pre["Pertenencia grupal del desaparecido"].value_counts())

data_pre.drop(columns = "Pertenencia grupal del desaparecido", inplace = True)
# Se elimina la columna

1     Persona adicta a una droga natural o sintética
2                                            Ninguno
4                                               Otro
5                                            Ninguno
6                                            Ninguno
8                                               Otro
9                                               Otro
10                                              Otro
11                                              Otro
12                                           Ninguno
Name: Pertenencia grupal del desaparecido, dtype: object
Pertenencia grupal del desaparecido
Ninguno                                                                           99669
Sin información                                                                   32333
Persona adicta a una droga natural o sintética                                    13279
Otro                                                                              10854
Campesinos (as) y/o traba

In [200]:
print(data_pre["Identidad de género del desaparecido"].head(10))
print(data_pre["Identidad de género del desaparecido"].value_counts())

data_pre["Identidad de género del desaparecido"].replace({
    "Transgénero": "No Aplica"
}, inplace = True)

data_pre["Identidad de género del desaparecido"].replace({
    "No Aplica": "LGBTIQ+"
}, inplace = True)
# Se agrupan las personas transgénero y las "No Aplica" como LGBTIQ+

1     No Sabe / No Informa
2     No Sabe / No Informa
4     No Sabe / No Informa
5     No Sabe / No Informa
6     No Sabe / No Informa
8     No Sabe / No Informa
9     No Sabe / No Informa
10    No Sabe / No Informa
11    No Sabe / No Informa
12    No Sabe / No Informa
Name: Identidad de género del desaparecido, dtype: object
Identidad de género del desaparecido
No Sabe / No Informa    156786
Masculino                11246
Femenino                  3341
No Aplica                 1556
Transgénero                 23
Name: count, dtype: int64


C:\Users\migue\AppData\Local\Temp\ipykernel_24272\384159287.py:4: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data_pre["Identidad de género del desaparecido"].replace({


In [201]:
print(data_pre["Orientación sexual del desaparecido"].head(10))
print(data_pre["Orientación sexual del desaparecido"].value_counts())

data_pre.drop(columns = "Orientación sexual del desaparecido", inplace = True)
# Se elimina la columna

1     No Sabe / No Informa
2     No Sabe / No Informa
4     No Sabe / No Informa
5     No Sabe / No Informa
6     No Sabe / No Informa
8     No Sabe / No Informa
9     No Sabe / No Informa
10    No Sabe / No Informa
11    No Sabe / No Informa
12    No Sabe / No Informa
Name: Orientación sexual del desaparecido, dtype: object
Orientación sexual del desaparecido
No Sabe / No Informa    161846
Heterosexual              9329
No Aplica                 1555
Homosexual                 159
Bisexual                    60
Asexual                      3
Name: count, dtype: int64


In [202]:
print(data_pre["Transgénero"].head(10))
print(data_pre["Transgénero"].value_counts())

data_pre.drop(columns = "Transgénero", inplace = True)
# Se elimina la columna

1     No Sabe / No Informa
2     No Sabe / No Informa
4     No Sabe / No Informa
5     No Sabe / No Informa
6     No Sabe / No Informa
8     No Sabe / No Informa
9     No Sabe / No Informa
10    No Sabe / No Informa
11    No Sabe / No Informa
12    No Sabe / No Informa
Name: Transgénero, dtype: object
Transgénero
No Sabe / No Informa    171392
No Aplica                 1555
Transgénero                  3
Transexual                   2
Name: count, dtype: int64


In [203]:
print(data_pre["Contexto"].head(10))
print(data_pre["Contexto"].value_counts())

data_pre.drop(columns = "Contexto", inplace = True)
# Se elimina la columna

1     Desaparición
2     Desaparición
4     Desaparición
5     Desaparición
6     Desaparición
8     Desaparición
9     Desaparición
10    Desaparición
11    Desaparición
12    Desaparición
Name: Contexto, dtype: object
Contexto
Desaparición    172952
Name: count, dtype: int64


En este proceso de limpieza y preparación se realizaron los siguientes cambios en las columnas de Dataframe:
- ID: Se observa que la columna "ID" contiene números consecutivos del 1 al 195808 y ninguno de estos valores se repite. Sin embargo, estos datos se pueden obtener el índice del Dataframe. Además, al eliminar y/o filtrar algunos datos este "ID" no va a aportar información significativa.
- Fecha de la desaparición: Se observa que esta columna contiene la fecha de la desaparición seguida del carácter "T00:00:00.000". El procedimiento a desarrollar es eliminar este caracter y convertir la fecha a tipo Date.
- Año de la desaparición: En esta columna hacen falta algunos años. Por lo tanto, usamos la columna fecha para obtener este dato. Además, filtramos el Dataframe entre los años 1986 y 2024.
- Mes de la desaparición: Se considera que, aunque la columna aporta el nombre del mes, este puede ser obtenido fácilmente de la columna "Fecha de la desaparición" por lo que se procede a eliminar.
- Dia de la desaparición: Esta columna contiene el dia de la semana que desapareció el individuo. Sin embargo, se observa que algunos datos (129) no contienen información por lo que, usando la columna "Fecha de la desaparición", se extrae el dia de la semana que ocurrió el hecho. Esta información se considera interesante para observar si hay alguna tendencia respecto al dia de la semana que desaparece una persona.
- Estado de la desaparición: En esta columna aparece si el individuo Apareció vivo, muerto o sigue desaparecido. Aqui solamente se reemplazan los valores "Aparecio Vivo" y "Aparecio Muerto" por "Aparecido Vivo" y "Aparecido Muerto", respectivamente, para mejor claridad de la información.
- Clasificación de la desaparición: Esta columna se conserva ya que se realizará un análisis general de desaparecidos y un análisis de desapariciones presuntamente forzosas.
- Entidad que realiza el registro de la desaparición: Esta columna no se considera relevante para el análisis por lo que se procede a eliminar.
- País donde ocurre la desaparición: En esta columna se conservan los registros de los individuos desaparecidos en Colombia. Al filtrar para Colombia el Dataset disminuye de 189934 a 188639 registros. Luego, se elimina la columna.
- Departamento donde ocurre la desaparición DANE: Esta columna se conserva ya que aporta información de interés de los departamentos donde se han reportado desapariciones.
- Codigo Dane Departamento: Los datos de esta columna se utilizarán para realizar un mapa donde se identificarán el número de desapariciones por departamento por periodo presidencial. Se cambia a un dato de tipo entero.
- Municipio donde ocurre la desaparición DANE: Esta columna se conserva para hacer un análisis de los municipios con más desapariciones en colombia.
- Codigo Dane Municipio: Esta columna se elimina ya que para el análisis basta con saber el nombre del municipio.
- Zona donde ocurre la desaparición: Esta columna se conserva para el análisis.
- Localidad donde ocurre la desaparición: Se elimina esta columna ya que no se realizará un análisis de desapariciones en Bogotá.
- Sexo del desaparecido: Se conserva para el análisis.
- País de nacimiento del desaparecido: Se filtra el Dataset por "colombia" y se elimina esta columna. Al filtrar el Dataset Dsiminuye de 188639 a 172952 registros.
- Estado civil del desaparecido: Esta columna se considera útil para el análisis, por lo tanto, se conserva.
- Grupo mayor y menor de edad del desaparecido: Se elimina porque no se conserva relevante para el análisis.
- Grupo de edad quinquenal del desaparecido: Se considera que tiene información muy parecida a las columnas "Grupo mayor y menor de edad del desaparecido" y "Grupo de edad judicial del desaparecido". Por ende, se elimina esta columna.
- Grupo de edad judicial del desaparecido: Se conserva esta columna llevando todas las edades a un formato común.
- Ciclo vital del desaparecido: Se conserva ya que puede realizarse un análisis al rededor de esto.
- Escolaridad del desaparecido: De esta columna se puede extraer un análisis, por lo tanto, se conserva.
- Pertenencia étnica del desaparecido: Esta columna se conserva ya que se considera de interés conocer esta variable en las desapariciones.
- Pueblo indígena del desaparecido: Se elimina esta columna ya que hay muy pocos datos de interés.
- Pertenencia grupal del desaparecido: Esta columna se elimina ya que no se realizará un análisis relacionado a este tema.
- Identidad de género del desaparecido: Se agrupan los "Transexuales" y "No Aplica" como LGBTIQ+.
- Orientación sexual del desaparecido: Se elimina esta columna ya que no hay suficientes datos relevantes para realizar este análisis.
- Transgénero: Se elimina ya que no hay suficiente información de la cual extraer un análisis.
- Contexto: Esta columna contiene un solo dato durante todo el Dataset. Al no ofrecer nueva información se elimina esta columna.

In [204]:
def calcularPeriodoPres(x):
    if (x >= 1986) and (x < 1990):
        return "1986-1990 | Virgilio Barco"
    elif (x >= 1990) and (x < 1994):
        return "1990-1994 | César Gaviria"
    elif (x >= 1994) and (x < 1998):
        return "1994-1998 | Ernesto Samper"
    elif (x >= 1998) and (x < 2002):
        return "1998-2002 | Nicolás Pastrana"
    elif (x >= 2002) and (x < 2006):
        return "2002-2006 | Álvaro Uribe 1"
    elif (x >= 2006) and (x < 2010):
        return "2006-2010 | Álvaro Uribe 2"
    elif (x >= 2010) and (x < 2014):
        return "2010-2014 | Juan Manuel Santos 1"
    elif (x >= 2014) and (x < 2018):
        return "2014-2018 | Juan Manuel Santos 2"
    elif (x >= 2018) and (x < 2022):
        return "2018-2022 | Iván Duque"
    elif (x >= 2022) and (x <= 2024):
        return "2022-2024 | Gustavo Petro"
# Se crea una función para calcular el periodo presidencial

data_pre["Periodo presidencial"] = data_pre["Año de la desaparición"].apply(lambda x: calcularPeriodoPres(x))
# Se aplica la función a la columna "Periodo presidencial"
data_pre["Periodo presidencial"].value_counts()

Periodo presidencial
2010-2014 | Juan Manuel Santos 1    32023
2014-2018 | Juan Manuel Santos 2    29044
2018-2022 | Iván Duque              23631
2002-2006 | Álvaro Uribe 1          19284
2006-2010 | Álvaro Uribe 2          19065
2022-2024 | Gustavo Petro           16975
1998-2002 | Nicolás Pastrana        16155
1994-1998 | Ernesto Samper           7956
1990-1994 | César Gaviria            4865
1986-1990 | Virgilio Barco           3954
Name: count, dtype: int64

Se creó una columna llamada "Periodo presidencial" para realizar el análisis en base a estos periodos.

In [205]:
size = data_pre.shape
data_pre.index = range(0, size[0])
# Se crea un índice nuevo para el Dataframe

In [206]:
data_pre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 172952 entries, 0 to 172951
Data columns (total 17 columns):
 #   Column                                          Non-Null Count   Dtype         
---  ------                                          --------------   -----         
 0   Fecha de la desaparición                        172952 non-null  datetime64[ns]
 1   Año de la desaparición                          172952 non-null  int64         
 2   Día de la desaparición                          172952 non-null  object        
 3   Estado de la desaparición                       172952 non-null  object        
 4   Clasificación de la desaparición                172952 non-null  object        
 5   Departamento donde ocurre la desaparición DANE  172952 non-null  object        
 6   Codigo Dane Departamento                        172952 non-null  int64         
 7   Municipio donde ocurre la desaparición DANE     172952 non-null  object        
 8   Zona donde ocurre la desaparición 

Luego de la limpieza el Dataframe este quedó de 17 columnas y 172952 registros (filas)


### 4. Manipulación y carga de datos

Se carga el Dataaframe limpio en un archivo plano CSV.

In [207]:
data_pre.to_csv("Datasets/desaparecidos_limpio.csv")